### ISSUE TITLE
Don't bind units on "ulocate" and "within"

### ISSUE THREAD

**Issue by limonovthesecond2** [Anuken/Mindustry-Suggestions#4871]
### Describe what you would like changed, and why.

There are logics that simply support the players. For example, they can send units for repairs or move them from reconstruction to the last player-controlled unit. The key issue here is the use of `ulocate` and `within` commands in the background, which causes them to become processor bound. After unbinding, these units will not be tied to the last player and will simply be idle somewhere.
I believe that they are similar to the `sensor` command and therefore the binding of units to logic after their execution can be removed

### Describe the changes you want to propose. Include possible alternatives.

Remove unit binding after `ulocate` and `within` commands

**Comment by JasonP01** [comment]
Ever heard of a fun little instruction called unbind? That's in ucontrol... and has its own description ingame

**Comment by limonovthesecond2** [comment]
> Ever heard of a fun little instruction called unbind? That's in ucontrol... and has its own description ingame

Of course, but unfortunately it does not solve the problem

> After `unbind`ing, these units will not be tied to the last player and will simply be `idle` somewhere.

https://github.com/Anuken/Mindustry-Suggestions/assets/118817903/f0de0f89-ebc4-4955-a4cd-fe29cdf19bb2

**Comment by JasonP01** [comment]
Ah command mode, i didnt pick up that detail in the previous comment. As for within, you could use sensor xy to do that (no ucontrol needed) as a solution for current version, for the ulocate implementation there is little you could do that wouldnt hinder the behaviour

**Comment by rekokoro** [comment]
ulocate is a command for processor to pass value to variable rather than making some changes for unit. The same is with unit within command. 

Imagine selecting and commanding flock of units while watching how they stop one by one just because somewhere processor are checking their location.

**Comment by limonovthesecond2** [comment]
> As for within, you could use sensor xy

Indeed, for `within` there is a workaround that literally does the same thing as `within`.

> the ulocate implementation there is little you could do that wouldnt hinder the behaviour

The key feature of `ulocate` is to determine the location of the building for each unit **separately**. The thing you can do without ruining the gameplay is to move all units to only **one** building. This is especially critical for repair turrets, when a unit can go to the one that is located at the end of the map, through the enemy’s walls, and not to the nearest one literally a few blocks away from the unit.
This is not to mention the fact that you also need to use `ulocate` to define it. Moreover, you should check from time to time with `ulocate` whether this building is still alive.
To sum it up, there are no bindingless alternatives to `ulocate`.

**Comment by rylid-umil** [comment]
GITHUB AT FAULT
